src/library/baseplaylistfeature.cpp

```diff
@@ -618,27 +618,26 @@ void BasePlaylistFeature::htmlLinkClicked(const QUrl& link) {
   * This method queries the database and does dynamic insertion
 */
 QModelIndex BasePlaylistFeature::constructChildModel(int selected_id) {
-    buildPlaylistList();
+    QList<IdAndLabel> playlistLabels = createPlaylistLabels();
     QList<TreeItem*> data_list;
     int selected_row = -1;
 
     int row = 0;
-    for (QList<QPair<int, QString> >::const_iterator it = m_playlistList.begin();
-         it != m_playlistList.end(); ++it, ++row) {
-        int playlist_id = it->first;
-        QString playlist_name = it->second;
+    for (auto it = playlistLabels.constBegin();
```

Comment:
Shorter and simpler variant:
```
for (const IdAndLabel& idAndLabel : createPlaylistLabels()) {
```
The actual container type doesn't matter for the iteration, so why depend on it.

I would even use `const auto&&` here and let the compiler do all the type checking. But I know that you prefer old-school explicit type declarations ;)

Reply:
But don't forget to the `row++` in the loop body. That's the only drawback.